### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#1886] [General]
## News

Hi everyone, the long awaited v2 release is finally here! With this release we're introducing many new features, a revamped look, and lots of bug fixes. We've made major changes to the underlying schema and libraries for improved performance and allows us to quickly add new features in the future.

### Improved location

In addition to country location, Umami now collects city and region information.

![image](https://user-images.githubusercontent.com/477975/232599010-d0eb3d94-049e-4e0a-8a63-97cdda584fa0.png)

### Clean URLs

Previously, Umami would include the query string as part of the page URL. Now we show only the path. Query parameters can still be found in the **Query parameters** section.

![image](https://user-images.githubusercontent.com/477975/232599724-a98d7ce1-f2c6-4b3a-943c-40da03caf26a.png)

### Page titles

Umami now records page titles.

![image](https://user-images.githubusercontent.com/477975/232599474-4d54b753-6312-41db-984f-53f60b1e3263.png)

### Teams

The new teams feature allows you to share websites with other users without having to create a Share URL. When a user joins a team, they can select which website they want to share with the team.

![image](https://user-images.githubusercontent.com/477975/232597791-6d81ce8b-b0a7-4ebb-9502-93628ed52350.png)

### Event data

The event data feature has been completely redesigned. We've updated the documentation on how it works:

- Track events, https://umami.is/docs/track-events
- Tracker functions, https://umami.is/docs/tracker-functions

There is no UI available in this release, but event data will be core to a lot of upcoming features. Stay tuned!

### Tracker

The tracker script has been refactored to be much more lightweight and performant. It can now also track outbound link clicks.

### Documentation

The documentation on the website has been updated with v2 information. See https://umami.is/docs.
The old docs are still available under https://umami.is/docs/v1/getting-started.

### New languages
- Sinhala
- Khmer
- Swiss German

## Migrating v1 to v2

Updating the application should be pretty straight-forward, but you will need to migrate your data in a separate step.
We've written a guide on how to migrate your install from v1 to v2. See https://umami.is/docs/migrate-v1-v2.

## Updates
- Upgraded Next.js to `13.2.3`
- Upgraded Prisma to `4.11.0`

A huge thanks to all the contributors on this release! @screendriver @franciscao633 @kaisteinke @BennyDeeDev @seanghay @briancao @BE-CH @vidschofelix @slawiko @d19dotca @enricopaulini @juangacovas @EffakT @ppnplus 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by ariaieboy** [comment]
Does it support client hints API ?

**Comment by brunoamaral** [comment]
I was trying to upgrade my docker container but can't seem to get past the migration.

```
docker exec -it umami yarn add @umami/migrate-v1-v2
docker exec -it umami npx @umami/migrate-v1-v2
``` 

Now I get the error:

```
umami    | Raw query failed. Code: `42601`. Message: `db error: ERROR: syntax error at or near "version"`
```

Am I missing a step because I run inside docker?

**Reply by brunoamaral** [reply]
For those with the same problem I was having, the solution was to run `docker compose run umami yarn add @umami/migrate-v1-v2` and then `docker compose run umami npx @umami/migrate-v1-v2`

**Comment by cocowalla** [comment]
The v1.x API included `POST /api/collect`, which could be used to send both page views AND events. In v2.x this endpoint is now `POST /api/send`, but the docs say:

> type: (string) event is currently the only type available

How can we send page view events via the API with 2.x please?

**Comment by ohrie** [comment]
Should I upgrade the app docker image first, than migrate the data or the other way?

**Reply by franciscao633** [reply]
migrate the data on 1.4 container then upgrade.

**Comment by tim-hub** [comment]
after migration, does it require to manually update `umami.js` to `script.js`?

Is there an automatical way, or 301 redirect?

**Comment by franciscao633** [comment]
Yes, the default tracker script has changed and is listed in breaking changes [here](https://umami.is/docs/migrate-v1-v2).

Alternatively you can just use the env var `TRACKER_SCRIPT_NAME ` to keep it umami.

**Comment by vencho-mdp** [comment]
Hi! Do you have an estimated deadline on when are we going to be able to see the events payloads?

**Comment by TonsiTT** [comment]
How to track outbound links? Its mentioned in the changelog but it doesn't show up when clicking on outbound links so is there a way to activate that setting or something?

**Reply by brunoamaral** [reply]
I don't recall any specific method to track outbound links, but as long as you add an event to the <a> tag, it should work.

`<a href data-umami-event="{event-name}">outlink</a>`

**Reply by TonsiTT** [reply]
It does show as an event but not to which outbound page it left to

**Reply by brunoamaral** [reply]
from the documentation, you can pass event data. https://umami.is/docs/track-events 

> You can optionally pass along event_data with the data-umami-event-* annotation.
> 
> data-umami-event="Signup button"
> data-umami-event-email="bob@aol.com"
> data-umami-event-id="123"
> The additonal properties will result in { email: 'bob@aol.com', id: '123' } being recorded with the Signup button name.
>
